lib/tesla/middleware/telemetry.ex

```diff
@@ -3,30 +3,111 @@ if Code.ensure_loaded?(:telemetry) do
     @behaviour Tesla.Middleware
 
     @moduledoc """
-    Send the request time and meta-information through telemetry.
+    Emits events using the `:telemetry` library to expose instrumentation.
 
     ### Example usage
     ```
     defmodule MyClient do
       use Tesla
 
-      plug Tesla.Middleware.Telemetry
+      plug Tesla.Middleware.Telemetry, prefix: [:my_client]
```

Comment:
I'd like to make it 100% clear that this is optional - let's move this to the list of options and leave the default example without `prefix`

Reply:
Fair. I added a note on the option to help people understand when it's useful to set.